mplc/partner.py

```diff
@@ -123,6 +124,36 @@ def shuffle_labels(self, proportion_shuffled):
         for i in idx:
             np.random.shuffle(self.y_train[i])
 
+    def duplicate_data(self, partner_copied, proportion):
+        if not 0 <= proportion <= 1:
+            raise ValueError(
+                f"The proportion of labels to corrupted was {proportion} but it must be between 0 and 1."
+            )
+        # Select the indices where the label will be off-set
+        n = int(len(self.y_train) * proportion)
+
+        if len(partner_copied.x_train) > n:
+            idx = sample(list(range(len(self.y_train))), n)
```

Comment:
It is a bit weird to do the test on `x_train` and then apply something on `y_train`. It is known that they have the same length, but still